pkg/controllers/cert/cert_rotating_controller.go

```diff
@@ -0,0 +1,287 @@
+	c.ClusterClient, err = c.ClusterClientSetFunc(cluster.Name, c.Client, c.ClusterClientOption)
+	if err != nil {
+		klog.Errorf("Failed to create a ClusterClient for the given member cluster: %v, err is : %v", cluster.Name, err)
+		return controllerruntime.Result{}, fmt.Errorf("failed to create a ClusterClient: %w", err)
+	}
+
+	secret, err := c.ClusterClient.KubeClient.CoreV1().Secrets("karmada-system").Get(context.TODO(), "karmada-kubeconfig", metav1.GetOptions{})
+	if err != nil {
+		return controllerruntime.Result{}, fmt.Errorf("failed to get karmada kubeconfig: %w", err)
+	}
+
+	err = c.syncCertRotating(secret)
+	if err != nil {
+		return controllerruntime.Result{Requeue: true}, err
+	}
+	// msg := fmt.Sprintf("Sync work (%s) to cluster(%s) successful.", work.Name, clusterName)
+	// klog.V(4).Infof(msg)
+	// c.EventRecorder.Event(work, corev1.EventTypeNormal, workv1alpha1.EventReasonSyncWorkSucceed, msg)
+	return controllerruntime.Result{}, nil
+}
+
+func (c *CertRotatingController) syncCertRotating(secret *corev1.Secret) error {
+	if secret.Data == nil {
+		return fmt.Errorf("no client certificate found in secret %q", secret.Namespace+"/"+secret.Name)
+	}
+
+	karmadaKubeconfigData, ok := secret.Data["karmada-kubeconfig"]
+	if !ok {
+		return fmt.Errorf("no karmada kubeconfig found in secret %q", secret.Namespace+"/"+secret.Name)
+	}
+
+	config, err := clientcmd.Load(karmadaKubeconfigData)
+	if err != nil {
+		return err
+	}
+
+	clusterName := config.Contexts[config.CurrentContext].AuthInfo
+
+	certData := config.AuthInfos[clusterName].ClientCertificateData
+
+	shouldRotate, err := shouldRotateCert(certData)
+
+	if !shouldRotate {
+		return nil
+	}
+
+	// create a new private key
+	keyData, err := keyutil.MakeEllipticPrivateKeyPEM()
+	if err != nil {
+		return err
+	}
+
+	privateKey, err := keyutil.ParsePrivateKeyPEM(keyData)
+	if err != nil {
+		return fmt.Errorf("invalid private key for certificate request: %w", err)
+	}
+
+	oldCert, err := certutil.ParseCertsPEM(certData)
+	if err != nil {
+		return fmt.Errorf("unable to parse old certificate: %w", err)
+	}
+
+	csrData, err := certutil.MakeCSR(privateKey, &oldCert[0].Subject, nil, nil)
+	if err != nil {
+		return fmt.Errorf("unable to generate certificate request: %w", err)
+	}
+
+	csrName := clusterName + "-" + k8srand.String(5)
+
+	certExpirationSeconds := int32(oldCert[0].NotAfter.Second() - oldCert[0].NotBefore.Second())
+
+	certificateSigningRequest := &certificatesv1.CertificateSigningRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: csrName,
+		},
+		Spec: certificatesv1.CertificateSigningRequestSpec{
+			Request:           csrData,
+			SignerName:        SignerName,
+			ExpirationSeconds: &certExpirationSeconds,
+			Usages: []certificatesv1.KeyUsage{
+				certificatesv1.UsageDigitalSignature,
+				certificatesv1.UsageKeyEncipherment,
+				certificatesv1.UsageClientAuth,
+			},
+		},
+	}
+
+	err = c.Client.Create(context.TODO(), certificateSigningRequest)
```

Comment:
`CSR` resource is create in default namespace, maybe set namesapce prameter is better?